packages/api/test/e2e/promise-alex.spec.ts

```diff
@@ -64,29 +65,18 @@ describe.skip('alex queries', () => {
       console.log('api.derive.staking.controllers:', JSON.stringify(res));
 
       done();
-    });
+    }).catch();
   });
 
-
   it('makes a query at a latest block (specified)', async () => {
-    const header = await api.rpc.chain.getHeader();
-    const events = await api.query.system.events.at(header.hash);
-
-    events.forEach(({ event: { data, method, section }, phase, topics }, index) => {
-      console.error(index, phase.toString(), `: ${section}.${method}`, data.toString(), topics.toString());
-    });
+    const header: any = await api.rpc.chain.getHeader();
```

Comment:
`chain.getHeader() as Header`

Reply:
Yeah, I had that, but I stumbled upon quite a few problems with `Codec` which was the reason for ending up choosing `any` just because of the sheer number of tslint errors I was getting.

Reply:
The tricky bits are the derives - the actual derive values we have (see eg. DeriveBalances) are not Codec. So for those, they need to be casted (with a relevant FIXME). We even atm, do this internal to derive, drives me up the wall. If I had $0.02 for each time I've done that, well...

The `api.rpc` and `api.query` are actually fine - they return valid `Codec` types. So a simple `as <Type>` works in those cases.

Reply:
I ran into quite a few of these cases, so as a general question addressing all of them:

Would you declare `const header = ...` like this:
`const header: Header = await api.rpc.chain.getHeader() as Header;`
or like this:
`const header = await api.rpc.chain.getHeader() as Header;`
because it's already explicitly declared ` as Header` anyways?